conan/tools/cmake/cmakedeps/templates/macros.py

```diff
@@ -29,8 +29,10 @@ def template(self):
        macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS FRAMEWORKS_DIRS)
            if(APPLE)
                foreach(_FRAMEWORK ${FRAMEWORKS})
+                   unset(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND CACHE)
+
                    # https://cmake.org/pipermail/cmake-developers/2017-August/030199.html
-                   find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAMES ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS} CMAKE_FIND_ROOT_PATH_BOTH)
+                   find_library(CONAN_FRAMEWORK_${_FRAMEWORK}_FOUND NAMES ${_FRAMEWORK} PATHS ${FRAMEWORKS_DIRS} NO_CMAKE_PATH CMAKE_FIND_ROOT_PATH_BOTH)
```

Comment:
In the normal library find, we are doing:
```cmake
find_library(CONAN_FOUND_LIBRARY NAMES ${_LIBRARY_NAME} PATHS ${package_libdir}
                            NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
```
That is, using the global NO_DEFAULT_PATH to exclude all of them, including NO_CMAKE_PATH. Maybe we want to do exactly the same here?
And also to switch and use NO_CMAKE_FIND_ROOT_PATH instead?

Reply:
Hello @memsharded,

Thank you very much for your reply.
I've tested all the options listed under [find_library](https://cmake.org/cmake/help/latest/command/find_library.html), in their hierarchical order. This is the result that I observed:
1. NO_DEFAULT_PATH ✅
2. NO_PACKAGE_ROOT_PATH ❌
3. NO_CMAKE_PATH ✅
4. NO_CMAKE_ENVIRONMENT_PATH ❌
5. NO_SYSTEM_ENVIRONMENT_PATH ❌
6. NO_CMAKE_SYSTEM_PATH ❌
where ❌ - still broken & ✅ - behavior fixed

I also just tested `NO_CMAKE_FIND_ROOT_PATH` and _on it's own_ it doesn't make a difference.

As `NO_DEFAULT_PATH` disables everything, it works. I went for `NO_CMAKE_PATH` in the PR because I tried to find the lowest scope that fixes the problem.

To sum up: the combination of `NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH` seems to work, and I additionally tested it on a real Apple XCFramework. Let's see if it doesn't break any of your tests.

Reply:
Then, it looks good, we can change it to that combination to be consistent with the other.
If you want to be a bit more conservative, I am fine with doing the minimal change in ``cmake_find_package``, but ``CMakeDeps`` can be changed to align completely with the other find_library.
Thanks!

Reply:
Hello @memsharded,

I rebased and updated my two commits with `NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH` to ensure consistency with the other functions in the codebase.
Thanks for all the feedback and I hope that the PR can now be merged!